Issue #211: Remove "x-translate-status-4xx: 200" in HookHandler

oli-h opened the pull request:
- update to vertx-rest-storage 2.5.2 which has config-option to answer "200 OK" when DELETE a non-existing resource
- adapt to new Config-APU of vertx-rest-storage
- remove the unconditional "x-translate-status-4xx: 200"

see #211 for details. solves #211

mcweba commented:
Some "hook" unit tests always fail on travis:

```
[ERROR] Errors: 
[ERROR]   ListenerTest.testDeadLock:580 Â» ConditionTimeout Lambda expression in org.swis...
[ERROR]   ListenerTest.testRequestForwardingForTwoListenerAtDifferentResource:311->checkGETBodyWithAwait:775 Â» ConditionTimeout
[ERROR]   ListenerTest.testRequestForwardingForTwoListenerAtSameResource:217->checkGETBodyWithAwait:775 Â» ConditionTimeout
```
Can you please verify whether they work locally?

oli-h commented:
Updated PullRequest. Fixed tests which really failed because of the now removed "x-translate-status-4xx: 200" header.
But: those tests were shaky as they placed listener-hooks for http-GET requests - which is strange and probably unwanted. And a further bug prevented to set list of wanted http-methods while placing hooks within the tests
